### ISSUE TITLE
Doesn't work

### ISSUE THREAD

**Issue by travankor** [soreau/wayland-logout#5]
Using the C version from wf-shell 0.6.1 along with Wayfire 0.6

strace:
```
execve("/bin/wayland-logout", ["wayland-logout"], 0x7ffd78e3dfe0 /* 52 vars */) = 0
brk(NULL)                               = 0x56040edf5000
access("/etc/ld.so.preload", R_OK)      = 0
openat(AT_FDCWD, "/etc/ld.so.preload", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=0, ...}) = 0
close(3)                                = 0
openat(AT_FDCWD, "/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=97139, ...}) = 0
mmap(NULL, 97139, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f1daf402000
close(3)                                = 0
openat(AT_FDCWD, "/usr/lib/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0o\2\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=18395656, ...}) = 0
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f1daf400000
mmap(NULL, 1844152, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f1daf23d000
mprotect(0x7f1daf262000, 1654784, PROT_NONE) = 0
mmap(0x7f1daf262000, 1347584, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x25000) = 0x7f1daf262000
mmap(0x7f1daf3ab000, 303104, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x16e000) = 0x7f1daf3ab000
mmap(0x7f1daf3f6000, 24576, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1b8000) = 0x7f1daf3f6000
mmap(0x7f1daf3fc000, 13240, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f1daf3fc000
close(3)                                = 0
arch_prctl(ARCH_SET_FS, 0x7f1daf401540) = 0
mprotect(0x7f1daf3f6000, 12288, PROT_READ) = 0
mprotect(0x56040dd38000, 4096, PROT_READ) = 0
mprotect(0x7f1daf442000, 4096, PROT_READ) = 0
munmap(0x7f1daf402000, 97139)           = 0
socket(AF_UNIX, SOCK_STREAM, 0)         = 3
connect(3, {sa_family=AF_UNIX, sun_path="/run/user/1000/wayland-1"}, 110) = 0
getsockopt(3, SOL_SOCKET, SO_PEERCRED, {pid=20451, uid=1000, gid=1000}, [12]) = 0
kill(20451, SIGINT)                     = 0
exit_group(0)                           = ?
+++ exited with 0 +++
```

**Comment by soreau** [comment]
> kill(20451, SIGINT)                     = 0

It is sending SIGINT signal to pid 20451, can you verify this is the pid of wayfire?

**Comment by travankor** [comment]
Yes, that's the correct pid.

**Comment by soreau** [comment]
If you run `kill` from the shell with the pid, does wayfire exit?

**Comment by travankor** [comment]
yes

**Comment by soreau** [comment]
I installed wayfire 0.6.0 and wf-shell 0.6.1 built with -Dwayland-logout:implementation=c and it works fine here. Does it work if you use the shell implementation of wayland-logout?

**Comment by travankor** [comment]
No, that doesn't work. Seems like SIGINT doesn't kill the process for me?

**Comment by travankor** [comment]
I'm running from a launcher script -- maybe that's why SIGINT won't work?

**Comment by epsilon-0** [comment]
> Seems like SIGINT doesn't kill the process for me?

I think you may have bigger problems in your computer...

**Comment by soreau** [comment]
But `kill -SIGINT $(pidof wayfire)` works? (and `pidof wayfire` is what you expect?)

**Comment by soreau** [comment]
> I'm running from a launcher script -- maybe that's why SIGINT won't work?

I run wayfire from a script as well.

**Comment by soreau** [comment]
Since `kill -INT` doesn't work either as discussed on IRC, closing this because wayland-logout is working correctly. A wrapper script to start wayfire is to blame but no further details yet.

**Comment by soreau** [comment]
TL;DR: Backgrounding processes with & makes kill -INT not work. We now use SIGTERM.
